frontend/src/app/navigation/ExtensibleNav.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { Nav, NavList } from '@patternfly/react-core';
+import {
+  isNavExtension,
+  isNavSectionExtension,
+  NavExtension,
+} from '@odh-dashboard/plugin-core/extension-points';
+import { useExtensions } from '@odh-dashboard/plugin-core';
+import { NavItem } from './NavItem';
+import { compareNavItemGroups } from './utils';
+
+type Props = {
+  label: string;
+};
+
+export const ExtensibleNav: React.FC<Props> = ({ label }) => {
+  const extensions = useExtensions<NavExtension>(isNavExtension);
+  const topLevelExtensions = React.useMemo(
+    () =>
+      extensions
+        .filter((e) => isNavSectionExtension(e) || !e.properties.section)
```

Comment:
```suggestion
  const extensions = useExtensions<NavExtension>(isNavSectionExtension);
```

Shouldn't you just directly pull the extensions you want?

Reply:
Top level items are not just sections but also links which are not part of a section such as the `Home` link.

It's certainly possible that I make a new type guard function and do it in one shot instead of getting the extensions and then filtering again.